eval/document.go

```diff
@@ -52,7 +53,7 @@ func (doc *Document) FetchData(ctx context.Context) (plugindata.Data, diagnostic
 	return result, diags
 }
 
-func (doc *Document) RenderContent(ctx context.Context, docDataCtx plugindata.Map) (plugin.Content, plugindata.Data, diagnostics.Diag) {
+func (doc *Document) RenderContent(ctx context.Context, docDataCtx plugindata.Map, name string) (plugin.Content, plugindata.Data, diagnostics.Diag) {
```

Comment:
Minor nitpick: I don't think that adding a `name` argument here is worth it just to display it in a diagnostic. 
I suggest adding the ability to refer back to the document block through our structures:

* add a `Source *definitions.ParsedDocument` to the `Document` struct [here](https://github.com/blackstork-io/fabric/blob/1aceed03feec09474928d932d1fdbe976dcfe58d/eval/document.go#L17), set it to `node` [here](https://github.com/blackstork-io/fabric/blob/1aceed03feec09474928d932d1fdbe976dcfe58d/eval/document.go#L138)
* add a `Raw *definitions.Document` [here](https://github.com/blackstork-io/fabric/blob/1aceed03feec09474928d932d1fdbe976dcfe58d/parser/definitions/parsed_document.go#L3), set it to `d` [here](https://github.com/blackstork-io/fabric/blob/1aceed03feec09474928d932d1fdbe976dcfe58d/parser/parse_document.go#L16)